peekingduck/training/test/test_training/tensorflow_training.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tensorflow test script"""
+
+import tensorflow as tf
+
+print("TensorFlow version:", tf.__version__)
+print("GPU:\n\n", tf.config.list_physical_devices("GPU"))
+
+sys_details = tf.sysconfig.get_build_info()
+print(sys_details)
+cuda_version = sys_details["cuda_version"]
+print("cuda version:", cuda_version)
+cudnn_version = sys_details["cudnn_version"]
+print("cudnn version:", cudnn_version)
+
+
+mnist = tf.keras.datasets.mnist
+
+(x_train, y_train), (x_test, y_test) = mnist.load_data()
+x_train, x_test = x_train / 255.0, x_test / 255.0
+
+model = tf.keras.models.Sequential(
+    [
+        tf.keras.layers.Flatten(input_shape=(28, 28)),
+        tf.keras.layers.Dense(128, activation="relu"),
+        tf.keras.layers.Dropout(0.2),
+        tf.keras.layers.Dense(10),
+    ]
+)
+
+predictions = model(x_train[:1]).numpy()
+
+tf.nn.softmax(predictions).numpy()
+
+loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+
+loss_fn(y_train[:1], predictions).numpy()
+
+model.compile(optimizer="adam", loss=loss_fn, metrics=["accuracy"])
+
+model.fit(x_train, y_train, epochs=5)
+
+model.evaluate(x_test, y_test, verbose=2)
+
+probability_model = tf.keras.Sequential([model, tf.keras.layers.Softmax()])
+
+probability_model(x_test[:5])
```

Comment:
Question: If this test runs successfully, it only proves that the Tensorflow training can run from start to end. But where is the test of correctness? There should be some kind of `assert(trained_model_inf_result == expected_inf_result)` towards the end to test correctness. Or am I missing something?

Reply:
`pytorch_training.py` and `tensorflow_training.py` without the `test_` prefix are used for internal testing on new machine. For example, to test whether the cuda library is installed correctly.

However, these to scripts are under `test` folder and pytest wouldn't run them without the `test_` prefix. May lead to some confusion on their purpose.